### PR TITLE
 Require setting a native vlan before adding any trunk vlans on the brocade switch (and correctly parse range of vlans)

### LINKED ISSUE #728: Brocade Driver: disconnecting a network that's attached natively does not work
Ran into this while testing #683. 

~~If a network is attached natively, then calling `node_detach_network` on that network does not actually~~ ~~remove it on the switch. ~~

When we remove a native network, the switchport defaults to VLAN 1 as it's native VLAN.

### PR DESCRIPTION
**Pull request by naved001**
* Moved that fix from the dellnos9 driver to switches.common and then used it
in brocade.

* Fixes #728 

It's a work in progress because `tests/deployment/vlan_networks.py` is failing. Gotta dig into that and push a fix asap.

### CODE REVIEW

File: hil/ext/switches/brocade.py [hunk 1]
```diff
@@ -48,6 +49,10 @@ def validate(kwargs):
     def session(self):
         return self
 
+    def ensure_legal_operation(self, nic, op_type, channel):
+        check_native_networks(nic, op_type, channel)
+        return
```
Comment (zenhack) [inline hil/ext/switches/brocade.py]: This return statement is superfluous

File: hil/ext/switches/dellnos9.py [hunk 2]
```diff
@@ -51,23 +51,8 @@ def session(self):
...
-        elif channel == 'vlan/native' and op_type == 'detach' and \
-                query.filter(table.channel != 'vlan/native').count() > 0:
-            # if it is detaching a network, then check in the database if there
-            # are any trunked vlans.
-            raise BlockedError("Please remove all trunked Vlans"
-                               " before removing the native vlan")
-        else:
-            return
+        check_native_networks(nic, op_type, channel)
+        return
```
Comment (zenhack) [inline hil/ext/switches/dellnos9.py]: Same thing here.

**Comment by naved001** [comment]
figured out why the deployment tests were failing. 

```
<hil.model.Port object at 0x7fd778e0e090>: [('vlan/native', '1514'), ('vlan/1511-1512', '1511-1512')]}
```
Now that I am attaching 3 networks to a port, the output is a range, which we don't parse correctly. Will push a fix soon.

**Comment by naved001** [comment]
okey dokey. The deployment tests now pass for brocade switch.

```
(.venv) naved:~/brocade-native-vlan/hil$ py.test tests/deployment/*networks.py
========================================= test session starts =========================================
platform linux2 -- Python 2.7.12, pytest-3.4.1, py-1.5.2, pluggy-0.6.0
rootdir: /home/naved/brocade-native-vlan/hil, inifile: setup.cfg
plugins: cov-2.5.1, forked-0.2, xdist-1.22.2
collected 2 items                                                                                     

tests/deployment/native_networks.py .                                                           [  0%]
tests/deployment/vlan_networks.py .

====================================== 2 passed in 53.40 seconds ======================================
```

**Comment by naved001** [comment]
coveralls doesn't seem to be happy, but I am already testing the stuff I need to :/

**Comment by naved001** [comment]
@zenhack Deployment tests passed with a caveat (unrelated to this PR):

for `vlan_networks.py` deployment tests if I specify a range of vlans in `testsuite.cfg` then the switch returns a [range of vlans](https://github.com/CCI-MOC/hil/pull/958#issuecomment-368948978) for which I have pushed a fixed in #958. The simple workaround was to specify non contiguous VLANs.

File: hil/ext/switches/common.py [hunk 3]
```diff
@@ -9,3 +12,46 @@ def should_save(switch_obj):
...
+        # checks if it is trying to attach a trunked network, and then in
+        # in the db see if nic does not have any networks attached natively
+        raise BlockedError("Please attach a native network first")
+    elif channel == 'vlan/native' and op_type == 'detach' and \
+            query.filter(table.channel != 'vlan/native').count() > 0:
+        # if it is detaching a network, then check in the database if there
+        # are any trunked vlans.
+        raise BlockedError("Please remove all trunked Vlans"
+                           " before removing the native vlan")
+    else:
```
Comment (Izhmash) [inline hil/ext/switches/common.py]: We don't really need the `else` or the `return` here.

Comment (naved001) [reply hil/ext/switches/common.py]: done

**Comment by zenhack** [comment]
LGTM, assuming travis patches I'll merge.

**Comment by naved001** [comment]
@zenhack @xuhang57  are we ignoring coveralls here too, or is there anything that I can do to make it happy.

**Comment by xuhang57** [comment]
Based on what the coverage tests are indicating, I would safe we should ignore the coveralls. We should merge.

**Comment by naved001** [comment]
I am gonna go ahead and merge this.

**Pull request merged**